### PR TITLE
lock the versions of a bunch of stuff used in the CI builds

### PR DESCRIPTION
**Pull request by greglandrum**
The CI builds run on PRs are not the place to find that things have changed in new releases of our dependencies

### CODE REVIEW

File: .azure-pipelines/linux_build.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ steps:
     conda create --name rdkit_build $(python) cmake \
         boost-cpp=$(boost_version) \
         py-boost=$(boost_version) \
-        numpy pillow eigen pandas matplotlib-base \
+        numpy pillow eigen pandas=2.0 matplotlib-base=3.7 \
```
Comment (bp-kelley) [inline .azure-pipelines/linux_build.yml:12]: You have pandas 2.1 everywhere else, is this intentional?

**Review by bp-kelley — CHANGES_REQUESTED** [review]

File: .azure-pipelines/linux_build.yml (hunk 1 above)
Comment (greglandrum) [reply .azure-pipelines/linux_build.yml:12]: > You have pandas 2.1 everywhere else, is this intentional?

yeah, 2.1 isn't available for python 3.7

**Review by bp-kelley — APPROVED** [review]

**Comment by bp-kelley** [comment]
Merged, I'll ping to failing PRs to merge from main

**Comment by greglandrum** [comment]
> Merged, I'll ping to failing PRs to merge from main

Thanks!

**Pull request merged**